source/core/CEntryPoints.cpp

```diff
@@ -199,81 +199,62 @@ VIREO_EXPORT const char* EggShell_ReadValueString(TypeManagerRef tm,
     }
     return "";
 }
+void CopyArrayTypeNameStringToBuffer(StringRef arrayTypeNameBuffer, SubString arrayTypeName)
+{
+    arrayTypeNameBuffer->Append(arrayTypeName.Length(), (Utf8Char*)arrayTypeName.Begin());
+    arrayTypeNameBuffer->Append((Utf8Char)'\0');
+}
+
+unsigned char* GetArrayBeginAt(TypedArrayCoreRef arrayObject)
+{
+    if (arrayObject->GetLength(0) <= 0) {
+        return null;
+    } else {
+        return arrayObject->BeginAt(0);
+    }
+}
 //------------------------------------------------------------
 //! Get information about an Array such as the type of its subtype, the array rank,
 //! and the memory location of the first element (or null if there are zero elements)
 VIREO_EXPORT EggShellResult EggShell_GetArrayMetadata(TypeManagerRef tm,
         const char* viName, const char* eltName, char** arrayTypeName, Int32* arrayRank, unsigned char** arrayBegin)
 {
-    TypeManagerScope scope(tm);
-    void *pData = null;
-
     SubString objectName(viName);
     SubString path(eltName);
+    void *pData = null;
+
     TypeRef pathType = tm->GetObjectElementAddressFromPath(&objectName, &path, &pData, true);
     if (pathType == null)
         return kEggShellResult_ObjectNotFoundAtPath;
 
-    if (!pathType->IsArray() || pathType->Rank() <= 0)
```

Comment:
Please elaborate why this check is no longer needed.

Reply:
Shouldn't the Rank() <= 0 check be moved to Data_GetArrayMetadata?

Reply:
Because it's checking the path and in Data_GetArrayMetadata we've already working with the array pointer instead of the path at this point, that check should indeed still exist but exist in the EggShell_GetArrayMetadata (which I am changing to be called EggShell_ArrayPointer)